polars/polars-lazy/polars-plan/src/dsl/meta.rs

```diff
@@ -22,6 +22,12 @@ impl MetaNameSpace {
     pub fn root_names(&self) -> Vec<Arc<str>> {
         expr_to_leaf_column_names(&self.0)
     }
+    /// A projection that only takes a column or a column + alias.
+    pub fn is_simple_projection(&self) -> bool {
+        let mut arena = Arena::with_capacity(8);
+        let node = to_aexpr(self.0.clone(), &mut arena);
```

Comment:
@ritchie46 is there any way to do this without cloning the expr?

Reply:
Yes, implementing the `is_simple_projection` for the `Expr` type. Maybe a good follow up PR?